### ISSUE TITLE
Links in confirmation emails not working for SOME users

### ISSUE THREAD

**Issue by NickWeir63** [openfoodfoundation/openfoodnetwork#2191]
When a new user registers they receive an email asking them to confirm their registration.  For most user this works fine.   However about 8 UK users have reported that the link in the email does not work for them.  

If I click on the link it works fine and does the job of confirming the registration.

This is a serious bug because we suspect that for every user who reports this issue there are many more who just give up and do not register.

This issue also happens for some users when they get the password reset email (the link doesnt work)

In t[his trello card](https://trello.com/c/kl97U8Df/153-links-in-emails-not-working) we have listed details of all the users who reported this issue.  Here is one example:
From: Jill Pidgeon [mailto:jillypidgeon@gmail.com]
Sent: 27 March 2018 11:07
To: nick@openfoodnetwork.org.uk
Subject: Re: Please confirm your OFN account

It was gmail
I was using and on my iPhone so not sure of anti virus software!

Sorry

Sent from my iPhone

On Tue, Mar 27, 2018 at 7:05 AM, Jill Pidgeon jillypidgeon@gmail.com wrote:
Hi there,

I am trying to register but when I click on the link to confirm my email it goes blank and says safari can't open the page... can you help? This is the third time since the weekend I ha e tried to open this!

Many thanks,

Jill

Sent from my iPhone

From: hello@openfoodnetwork.org.uk [mailto:hello@openfoodnetwork.org.uk]
Sent: 27 March 2018 07:04
To: jillypidgeon@gmail.com
Subject: Please confirm your OFN account

OPEN FOOD NETWORK UK

Hello! 
Welcome to Open Food Network UK!
Before we can activate your new account, we need to confirm your email address.
Please click the link below to confirm your email and to continue setting up your profile.
Confirm this email address »

Cheers,
Open Food Network UK Team

Connect with Us: 
Facebook Twitter

Email us: 
hello@openfoodnetwork.org.uk

Mar 27 at 9:07 AM - Edit - Add Link as Attachment - Delete - 

Sinead Fenton
Jill Pidgeon - 27/03 - confirmed by sinead on 27/03 by clicking this link sent to Jill on the 24th March:
https://u2720784.ct.sendgrid.net/wf/click?upn=xFBGS0monMFzX-2FqZT940KkIv9e1v2-2B4bVloQEQGLPesDxcfV-2F33pWU2RhxA0Qle6EH-2FcLNtgKtX-2FrGNLfFT3GyVbwL4vMEr5DnQBAfYc535PKbc2-2BrJ82G1DdT49EizL-2BDuW6IasGotXrwz5uILg0w-3D-3D_xrz9Osm27fmSTEjXaSqgiraY8dFtcPtl7Pt-2FAIpayk9-2F8P0UEIWcV-2BCcTLoGxLXJyszEVsVi1uZNyHPQE79wO4INsWpoy1yTD283Z-2Fwf5AEYsDm6jDys6E56yXGRa5Dpxg-2FGQWTFILa-2FTUGeMkPwZqo2gJNOTTgSZmqap2vZYqZLJOe642IzdtBnAkYyE7DwxsxKUwCpfIr28eaGmNPG7qmY4emJllDPxo3i6LSeLuE-3D

If you need access to the Trello card please ask @lin-d-hop as I am away until 7th APril.

Thanks for looking at this

**Comment by myriamboure** [comment]
Please folks we have set up a template for reporting issues would be great to follow it! @NickWeir63 when you create a new issue you have the template to fill, so you can just follow it. It makes dev work much more easier then to understand the problem. You can also set up a severity for the bug which then allow the pipeline team to prioritize and react accordingly. So in this case it is definitely a S2 as some users don't even manage to register on the platform.

**Comment by myriamboure** [comment]
@mkllnk @Matt-Yorkley @sauloperez do you have any clue where that could come from? Who wants to take that on? Can it be link more to sys admin UK things or does it sound more like code related? Seems like a pretty urgent s2 bug...

**Comment by Matt-Yorkley** [comment]
Does anyone have a Mac with Safari that could give this a quick test by signing up on UK production and clicking the validation link?

**Comment by kirstenalarsen** [comment]
just did it - worked just fine :)

**Comment by Matt-Yorkley** [comment]
Thanks @kirstenalarsen! It seems this has appeared at the same time as UK production SSL certificate issues, which may have been fixed now by @mkllnk? And from some of these email error reports it looks like SSL related issues regarding addresses not resolving properly were the cause of the blank screen... so I think this might be resolved, but we can keep an eye on it.

**Comment by myriamboure** [comment]
Ok I put that in testing then so that we can just wait a bit and observe, and we can close in a few days if all is green :-)

**Comment by lin-d-hop** [comment]
Just to note this definitely didn't correlate with SSL issues. This has been happening since v1.11 2 weeks ago. SSL was a problem for 3hrs this morning. The problem is not specific to OS or mail client.

**Comment by lin-d-hop** [comment]
I'm sorry I've had no capacity to look into it yet.

**Comment by mkllnk** [comment]
In Slack Lynne posted:

> This site can’t be reached
https’s server IP address could not be found.
Search Google for https openfoodnetwork org user spree password edit
ERR_NAME_NOT_RESOLVED

That sounds like the DNS server failed. Unfortunately, the emails above don't show the URL people tried to visit.

I can't see any problem in you DNS setup. Maybe your DNS server was just not reliable.

**Comment by sauloperez** [comment]
@mkllnk is right, we need the actual URL these users clicked on.

**Comment by mkllnk** [comment]
I guess they are "testing" the current code on UK production. But yes, In Dev is probably a better description.

**Comment by NickWeir63** [comment]
Sorry I have been away for a week and just catching up.  Below is one of the failing URL's - many of the other failing urls and automatic emails are in the Trello card where we have been collecting details of users who reported problems on this.    let me know if you want access to the Trello card.  thanks.
https://u2720784.ct.sendgrid.net/wf/click?upn=xFBGS0monMFzX-2FqZT940KkIv9e1v2-2B4bVloQEQGLPesDxcfV-2F33pWU2RhxA0Qle6EH-2FcLNtgKtX-2FrGNLfFT3GyVbwL4vMEr5DnQBAfYc535P20fIwbyrTvbyTEPfzvr0x5Powc4DG44ty6S2HQSOsg-3D-3D_CmwAv3oRa0AH4Hd1bWC6X4EQhJV0tgjocrFR9kc1S0Gt5rx2ZWBWqFEKHB2rukfwKxKxJk0k-2FvLQlrskbJZ-2FN5IpYh7GxiBGaceA8Q-2Fs-2Bo4IvbLbPSbTOTUzSoBsB8MN12kRdMhJpL3JD-2BURnkQfdCMQnsE9KXOxSvE4qHuyXtjLVYoWSP4ngyfKx4-2BjhaI0KJZWYrgrRKWN3nyP6OmVa50cR0k-2FHY0ZSdA1Mj2-2Fos4-3D

**Comment by enricostano** [comment]
This URL get:
```
< HTTP/1.1 302 Found
< Server: nginx
< Date: Mon, 09 Apr 2018 13:25:56 GMT
< Content-Type: text/html; charset=utf-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Location: http://openfoodnetwork.org.uk/user/spree_user/confirmation?confirmation_token=cBB0rLKFBHu_EY47qdkC
< X-Robots-Tag: noindex, nofollow
< 
```
We need to check both URL and token in `Location`

**Comment by mkllnk** [comment]
And that gets redirected to the https version of that URL. I'm wondering why it's not using https straight away, but that's not the problem here.

**Comment by NickWeir63** [comment]
Here is another link that did not work for the user but (like all the other issues) it worked (successfully confirmed their registration) for Sinead and I when we click on the link for them.
https://u2720784.ct.sendgrid.net/wf/click?upn=xFBGS0monMFzX-2FqZT940KkIv9e1v2-2B4bVloQEQGLPesDxcfV-2F33pWU2RhxA0Qle6EH-2FcLNtgKtX-2FrGNLfFT3GyVbwL4vMEr5DnQBAfYc535PKbc2-2BrJ82G1DdT49EizL-2BDuW6IasGotXrwz5uILg0w-3D-3D_xrz9Osm27fmSTEjXaSqgiraY8dFtcPtl7Pt-2FAIpayk9-2F8P0UEIWcV-2BCcTLoGxLXJyszEVsVi1uZNyHPQE79wO4INsWpoy1yTD283Z-2Fwf5AEYsDm6jDys6E56yXGRa5Dpxg-2FGQWTFILa-2FTUGeMkPwZqo2gJNOTTgSZmqap2vZYqZLJOe642IzdtBnAkYyE7DwxsxKUwCpfIr28eaGmNPG7qmY4emJllDPxo3i6LSeLuE-3D
Let me know if you want access to the Trello card with details of all the problem emails that have been reported

**Comment by mkllnk** [comment]
Is there an option in your Sendgrid account to deactivate URL rewriting and tracking? Maybe these Sendgrid URLs are causing an issue. We are sending direct links to the OFN.

**Comment by daniellemoorhead** [comment]
Where's this at everybody? @NickWeir63 has the issue been fixed? Can we consider this closed?

If yes, then @NickWeir63 can you please close this issue. If no, then let's get to figuring out a plan B 😀

**Comment by NickWeir63** [comment]
I have had no more reports of problems so will close.  Thanks all
